src/hooks/usePagination.tsx

```diff
@@ -0,0 +1,48 @@
+export const usePagination = (
+  currentPage: number,
+  postsTotal: number | undefined
+) => {
+  const pagesAmount = Math.ceil(Number(postsTotal) / 8);
```

Comment:
`8` should be replace with `perPage` from function argument. Also, your whole algorithm should work with dynamic per page

When you have a function with a lot of arguments, it's better to have a single argument that is object

```
interface UsePaginationProps {
  currentPage: number;
  totalItems: number;
  perPage: number
}

export const usePagination = ({}: UsePaginationProps) => ...
```